### PR TITLE
 update client design

### PR DESCRIPTION
**Pull request by tonyyang-svail**

### CODE REVIEW

File: doc/client_design_doc.md [hunk 1]
```diff
@@ -13,41 +17,102 @@ import sqlflow
 
 client = sqlflow.Client(server_url='localhost:50051')
 
-# Regular SQL
-response = client.execute('select * from table')
-for dataframe in response:
-    print(dataframe) # {'y': [4, 5, 6], 'x': [1, 2, 3]}
+# Query SQL
+rowsets = client.query('SELECT ... FROM ...')
+for rowset in rowset:
```
Comment (ktong) [inline doc/client_design_doc.md]: the query should return only one rowset unless it supports multiple sqls.

File: doc/client_design_doc.md [hunk 2]
```diff
@@ -13,41 +17,102 @@ import sqlflow
...
 
-`sqlflow.Client` uses grpc to contact the `sqlflowserver`. The service protocol is defined at [sqlflow.proto](https://github.com/wangkuiyi/sqlflowserver/blob/develop/sqlflow.proto).
+## Implementaion
 
 `sqlflow.Client.__init__` establishes a grpc stub/channel based on `server_url`.
 
-`sqlflow.Client.execute` takes a sql statement
-- if the statement is a regular SQL, it returns a generator which generates dataframes
-- if the statement is a extended SQL, it prints the log
+`sqlflow.Client.query` takes a sql statement and returns a `RowSet` object.
```
Comment (ktong) [inline doc/client_design_doc.md]: query must merge into one Rowset if server returns multiple frames

Comment (tonyyang-svail) [reply doc/client_design_doc.md]: Good point.

File: doc/client_design_doc.md (hunk 1 above)
Comment (tonyyang-svail) [reply doc/client_design_doc.md]: Good abstraction.

**Pull request merged**